Support OCPP 2.1 variables (#988)
EVerest/libocpp: doc/v201/ocpp_201_device_model_initialization.md

Note: OCPP requires EVSE and Connector numbering starting from 1 counting upwards.

Note: There should be no duplicate components or variables in the component config files.

## Required variables

> There are some required Variables, which can be found in the OCPP spec.  

Reviewer: ```suggestion
There are Variables that are defined as **required** in the OCPP specification. 
```

> Some `Variables` are only required if the `Component` is `Available`, for example `Reservation` and `Smart Charging`. 

Reviewer: ```suggestion
Some `Variables` are only required if the `Available` variable of the respective `Component` is present and configured to true. Examples for this are the `ReservationCtrlr` and the `SmartChargingCtrlr`.
```

> There are  some Components that are always required because that is how libocpp works: `AlignedDataCtrlr` and 

Reviewer: ```suggestion
There are some Components that are always required because they are required by libocpp: `AlignedDataCtrlr` and 
```

`SampledDataCtrlr`.  
When libocpp is started and initialized, all required Variables will be checked and an DeviceModelError is thrown if 

> one of the required Variables is not there.

Reviewer: ```suggestion
one of the required Variables is not present.
```
